File: server/models/commands.py
from typing import Annotated

from pydantic import Field, EmailStr

from server.models.common import CustomBaseModel
from server.settings import settings


class EmailedInvolvementSummaryRequest(CustomBaseModel):
    member_full_name: str
    preceding_months: Annotated[int, Field(ge=settings.min_preceding_months, le=settings.max_preceding_months)]


class EmailedInvolvementSummaryRequestResult(EmailedInvolvementSummaryRequest):
    masked_email_address: EmailStr

File: shared/types/dto/commands.ts
import z from "zod"
import {MIN_PRECEDING_MONTHS, MAX_PRECEDING_MONTHS} from "#shared/utils/constants"

export const EmailedInvolvementSummaryRequestSchema = z.object({
  memberFullName: z.string(),
  precedingMonths: z.number().min(MIN_PRECEDING_MONTHS).max(MAX_PRECEDING_MONTHS),
})
export type EmailedInvolvementSummaryRequest = z.infer<typeof EmailedInvolvementSummaryRequestSchema>

export interface EmailedInvolvementSummaryRequestResult extends EmailedInvolvementSummaryRequest {
  maskedEmailAddress: string | null
}
